sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/TypesafeBackendImpl.kt

```diff
@@ -22,32 +23,20 @@ internal class TypesafeBackendImpl(private val backend: Backend) : TypesafeBacke
     override val network: ZcashNetwork
         get() = ZcashNetwork.from(backend.networkId)
 
-    override suspend fun initAccountsTable(vararg keys: UnifiedFullViewingKey) {
-        val ufvks = Array(keys.size) { keys[it].encoding }
-        @Suppress("SpreadOperator")
-        backend.initAccountsTable(*ufvks)
-    }
-
-    override suspend fun initAccountsTable(
+    override suspend fun createAccountAndGetSpendingKey(
         seed: ByteArray,
-        numberOfAccounts: Int
-    ): List<UnifiedFullViewingKey> {
-        return DerivationTool.getInstance().deriveUnifiedFullViewingKeys(seed, network, numberOfAccounts)
-    }
-
-    override suspend fun initBlocksTable(checkpoint: Checkpoint) {
-        backend.initBlocksTable(
-            checkpoint.height.value,
-            checkpoint.hash,
-            checkpoint.epochSeconds,
-            checkpoint.tree
+        checkpoint: Checkpoint,
```

Comment:
Given that the only thing that you do with the `Checkpoint` is extract the tree state from it, I feel like the caller could be responsible for invoking the `checkpoint.treeState().encoded` dereference and this would then be more flexible; you don't need an actual checkpoint to initialize the wallet, though if you have a checkpoint you can certainly use that. We don't initialize the blocks table with a block hash from the checkpoint any longer, so we no longer verify that downloaded blocks chain from a checkpointed hash. I don't think that's a problem though; even if we did chain from a `Checkpoint` object, it would be possible to construct a malicious `Checkpoint` to chain from.

Reply:
I like that idea. The `createAccountAndGetSpendingKey` API changed.